subsys/cpp/cpp_virtual.c

```diff
@@ -19,7 +19,7 @@
  */
 void __cxa_pure_virtual(void)
 {
-	while (1) {
+	while (true) {
```

Comment:
does C++ define "true"? Might need to pull in an include somwhere

Reply:
```
FAILED: ccache /opt/sdk/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/xtensa-zephyr-elf/xtensa-zephyr-elf-gcc -DBUILD_VERSION=zephyr-v1.13.0-2185-g9d859ae904 -DKERNEL -D_FORTIFY_SOURCE=2 -D__ZEPHYR__=1 -I../../../../../../kernel/include -I../../../../../../arch/xtensa/include -I../../../../../../soc/xtensa/sample_controller -I../../../../../../soc/xtensa/sample_controller/include -I../../../../../../soc/xtensa/include -I../../../../../../include -I../../../../../../include/drivers -Izephyr/include/generated -isystem /opt/sdk/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/lib/xtensa-zephyr-elf/gcc/xtensa-zephyr-elf/6.2.0/include -isystem /opt/sdk/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/lib/xtensa-zephyr-elf/gcc/xtensa-zephyr-elf/6.2.0/include-fixed -I../../../../../../lib/libc/minimal/include -I../../../../../../tests/include -I../../../../../../tests/ztest/include -I/opt/sdk/zephyr-sdk-0.9.5/sysroots/xtensa-zephyr-elf/usr/include/arch/include -I/opt/sdk/zephyr-sdk-0.9.5/sysroots/xtensa-zephyr-elf/usr/include -Os -g -Wall -Wformat -Wformat-security -Wno-format-zero-length -imacros /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/out-2nd-pass/qemu_xtensa/tests/application_development/cpp/test/zephyr/include/generated/autoconf.h -ffreestanding -Wno-main --sysroot /opt/sdk/zephyr-sdk-0.9.5/sysroots/xtensa-zephyr-elf/usr -Werror -Wno-deprecated-declarations -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -Wno-pointer-sign -Wno-unused-but-set-variable -fno-reorder-functions -fno-defer-pop -Werror=implicit-int -Wpointer-arith -ffunction-sections -fdata-sections -mlongcalls -std=c99 -MD -MT zephyr/CMakeFiles/zephyr.dir/subsys/cpp/cpp_virtual.c.obj -MF zephyr/CMakeFiles/zephyr.dir/subsys/cpp/cpp_virtual.c.obj.d -o zephyr/CMakeFiles/zephyr.dir/subsys/cpp/cpp_virtual.c.obj   -c /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/subsys/cpp/cpp_virtual.c
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/subsys/cpp/cpp_virtual.c: In function __cxa_pure_virtual:
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/subsys/cpp/cpp_virtual.c:22:9: error: true undeclared (first use in this function)
  while (true) {
         ^~~~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/subsys/cpp/cpp_virtual.c:22:9: note: each undeclared identifier is reported only once for each function it appears in
ninja: build stopped: subcommand failed.
```

Think the problem is that this file is being compiled by the C and not C++ compiler (note filename extension)

Reply:
FWIW, the C++ language defines "true" as a builtin token.  In C, the official standard (for backward compatibility reasons) defines "_True" as the builtin, and you're supposed to include <stdbool.h> to get the symbol that everyone actually uses.